app/src/main/java/org/mozilla/fenix/session/HardwareAccelerationLifecycleCallbacks.kt

```diff
@@ -0,0 +1,49 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.session
+
+import android.app.Activity
+import android.app.Application
+import android.os.Build
+import android.os.Bundle
+import android.view.WindowManager.LayoutParams.FLAG_HARDWARE_ACCELERATED
+
+/**
+ * This ActivityLifecycleCallbacks implementation ensures that hardware acceleration is enabled for
+ * a Window when an Activity is created. On some devices we cannot enable UI hardware acceleration
+ * due to graphics driver issues. So we must globally disable hardware acceleration in
+ * AndroidManifest.xml, then enable it where possible on a per-window basis.
+ * See https://bugzilla.mozilla.org/show_bug.cgi?id=1609191
+ */
+@SuppressWarnings("EmptyFunctionBlock")
+class HardwareAccelerationLifecycleCallbacks : Application.ActivityLifecycleCallbacks {
+    val supportsHardwareAcceleration by lazy {
+        // List of boards on which we should keep hardware acceleration disabled.
+        // These were the boards with the highest number of crashes caused by hardware acceleration being enabled.
+        // There will be some affected devices with different boards which therefore may still crash,
+        // however the numbers should be low enough and the crashes infrequent enough for that to be okay.
+        val buggyBoards = listOf("msm8953", "msm8976", "msm8937", "msm8996", "sdm450_mh4x")
+
+        Build.BOARD !in buggyBoards
+    }
+
+    override fun onActivityCreated(activity: Activity, savedInstanceState: Bundle?) {
+        if (supportsHardwareAcceleration) {
+            activity.window.setFlags(FLAG_HARDWARE_ACCELERATED, FLAG_HARDWARE_ACCELERATED)
+        }
```

Comment:
@pocmo discovered an interesting comment on the [Chrome's implementation](https://source.chromium.org/chromium/chromium/src/+/master:chrome/android/java/src/org/chromium/chrome/browser/app/ChromeActivity.java;l=2452-2474;drc=4770965e365681d372956ba49c01a44a2abb9b9b). I think it's useful to add it here for references, and give it a try to see if dialogs have any issues, a way to test dialogs are trying to download a file, opening any page with js prompts like [alerts](https://www.w3schools.com/jsref/met_win_alert.asp)  [confirm](https://www.w3schools.com/jsref/met_win_confirm.asp) and others. 

> 
> 
>             // When HW acceleration is enabled manually for an activity, child windows (e.g.
>             // dialogs) don't inherit HW acceleration state. However, when HW acceleration is
>             // enabled in the manifest, child windows do inherit HW acceleration state. That
>             // looks like a bug, so I filed b/23036374
>             //
>             // In the meanwhile the workaround is to call
>             //   window.setWindowManager(..., hardwareAccelerated=true)
>             // to let the window know that it's HW accelerated. However, since there is no way
>             // to know 'appToken' argument until window's view is attached to the window (!!),
>             // we have to do the workaround in onAttachedToWindow()

Reply:
I tried to access the bug they mentioned on the on the comment https://issuetracker.google.com/issues/23036374 but it is not publicly available.

Reply:
Dialogs from alerts and and downloads look exactly the same to me in all 3 scenarios:
a) No patch - hardware acceleration enabled by default
b) With patch, unaffected device - hardware acceleration enabled programatically
c) With patch, affected device - hardware acceleration disabled

That's not to say the bug doesn't still exist - It could be the case that hardware acceleration is indeed not being enabled for the dialogs in case b). But I wouldn't think it matters much if those are our only uses of dialogs.

I did initially implement this by creating a base Activity class and overriding `onCreate()` there, as I didn't know that `ActivityLifecycleCallbacks` existed. But when I stumbled across it in the docs I thought it was better to use it, as it means we don't have to derive all of Fenix's activities from a single class. If we want to do what chrome does we'll need to derive every activity from a single one (or duplicate some code in every activity) that so we can override `Activity.onAttachedToWindow()`

Reply:
> Dialogs from alerts and and downloads look exactly the same to me in all 3 scenarios:
> a) No patch - hardware acceleration enabled by default
> b) With patch, unaffected device - hardware acceleration enabled programatically
> c) With patch, affected device - hardware acceleration disabled

I think we can keep the actual implementation and add the comment for future references.

> That's not to say the bug doesn't still exist - It could be the case that hardware acceleration is indeed not being enabled for the dialogs in case b). But I wouldn't think it matters much if those are our only uses of dialogs.

There are other usages of dialogs across Fenix, but the above are the main ones.


> I did initially implement this by creating a base Activity class and overriding onCreate() there, as I didn't know that ActivityLifecycleCallbacks existed. But when I stumbled across it in the docs I thought it was better to use it, as it means we don't have to derive all of Fenix's activities from a single class. If we want to do what chrome does we'll need to derive every activity from a single one (or duplicate some code in every activity) that so we can override Activity.onAttachedToWindow()

This is clean solution, but if it's strictly needed, having a based activity will not be a big issue as Fenix is following [a single activity approach](https://www.youtube.com/watch?v=2k8x8V77CrU) for navigation ([HomeActivity.kt](https://github.com/mozilla-mobile/fenix/blob/master/app/src/main/java/org/mozilla/fenix/HomeActivity.kt)), in most of the app.